lib/datasets.go

```diff
@@ -74,136 +74,62 @@ type GetParams struct {
-	}
-	return selector, format, nil
-}
+	if p.Selector == "body" {
+		if !p.All {
+			// ensure valid limit value
+			if p.Limit <= 0 {
+				p.Limit = 25
+			}
+			// ensure valid offset value
+			if p.Offset < 0 {
+				p.Offset = 0
+			}
 
-func arrayContains(subject []string, target string) bool {
-	for _, v := range subject {
-		if v == target {
-			return true
 		}
 	}
-	return false
 }
 
-// UnmarshalFromRequest implements a custom deserialization-from-HTTP request
-func (p *GetParams) UnmarshalFromRequest(r *http.Request) error {
-	mvars := mux.Vars(r)
-
-	if p == nil {
-		p = &GetParams{}
-	}
-
-	params := *p
-	if params.Ref == "" {
-		params.Ref = r.FormValue("ref")
+// Validate returns an error if GetParams fields are in an invalid state
+func (p *GetParams) Validate() error {
+	if !isValidSelector(p.Selector) {
+		return fmt.Errorf("could not parse request: invalid selector")
 	}
-
-	ref, err := dsref.Parse(params.Ref)
-	if err != nil {
-		return err
-	}
-	if ref.Username == "me" {
-		return fmt.Errorf("username \"me\" not allowed")
-	}
-
-	if sel, ok := mvars["selector"]; ok && params.Selector == "" {
-		selector, format, err := parseSelector(sel)
-		if err != nil {
-			return err
+	if p.Selector == "body" {
+		if !p.All && (p.Limit < 0 || p.Offset < 0) {
+			return fmt.Errorf("invalid limit / offset settings")
 		}
-		params.Selector = selector
-		params.Format = format
 	}
 
-	if params.Format == "" {
-		params.Format = r.FormValue("format")
-	}
+	return nil
+}
 
-	// This HTTP header sets the format to csv, and removes the json wrapper
-	if arrayContains(r.Header["Accept"], "text/csv") {
-		if params.Format != "" && params.Format != "csv" {
-			return fmt.Errorf("format %q conflicts with header \"Accept: text/csv\"", params.Format)
-		}
-		params.Format = "csv"
-		params.Selector = "body"
+func isValidSelector(selector string) bool {
+	if selector == "" {
+		return true
 	}
 
-	if params.Format != "" && params.Format != "json" && params.Format != "csv" && params.Format != "zip" {
-		return fmt.Errorf("invalid extension format")
+	match := validSelector.FindString(selector)
+	if match == "" || len(match) != len(selector) {
+		return false
 	}
-
-	// TODO(arqu): we default to true but should implement a guard and/or respect the page params
-	params.All = true
-	// listParams := ListParamsFromRequest(r)
-	// offset := listParams.Offset
-	// limit := listParams.Limit
-	// if offset == 0 && limit == -1 {
-	// 	params.All = true
-	// }
-
-	*p = params
-	return nil
+	return true
 }
 
+var validSelector = regexp.MustCompile(`^[\w-\.]*[\w]$`)
+
 // GetResult combines data with it's hashed path
 type GetResult struct {
-	Ref       *dsref.Ref       `json:"ref"`
-	Dataset   *dataset.Dataset `json:"data"`
-	Bytes     []byte           `json:"bytes"`
-	Message   string           `json:"message"`
-	FSIPath   string           `json:"fsipath"`
-	Published bool             `json:"published"`
```

Comment:
Want to note that we are losing these two pieces of information, and need an alternative way of getting the `FSIPath` and the `Published` status. Although, thinking back to the ui presentation this morning, neither piece of information was needed in those mockups.